src/main/java/graphql/TypeResolutionEnvironment.java

```diff
@@ -21,7 +22,7 @@
 public class TypeResolutionEnvironment {
 
     private final Object object;
-    private final ImmutableMapWithNullValues<String, Object> arguments;
+    private final Supplier<ImmutableMapWithNullValues<String, Object>> arguments;
```

Comment:
This is a viral change - type resolving wont materialise the ESI arguments unless it has to